src/state_machine.cpp

```diff
@@ -22,96 +18,127 @@ void StateMachine::construct()
   State& sit_start = this->createGaitState(SIT, SIT_SELECTED, SIT_ACTIVATED, "gait_sit", &turn_off_start);
 
   // Sub-menus
-  this->constructObstacleMenu(&obstacles, &walk, &single_step, &obstacles);
-  this->constructWalkMenu(&walk, &obstacles);
-  this->constructStepMenu(&single_step, &obstacles);
+  this->constructObstacleMenu(&obstacles);
+
+  // Determine the last added state before the walk and single step menu are added
+  // This is used in constructing the walk and single step menu
+  std::list<State>::iterator last_state = states_.end();
+
+  this->constructWalkMenu(last_state);
+  this->constructStepMenu(last_state);
 
   // Menu transitions
-  home_stand_start.withRight(&home_sit_start)
-      .withLeft(&turn_off_start)
-      .shortcutPushTo(&walk)
-      .shortcutDoublePushTo(&single_step);
-  home_sit_start.withRight(&stand_up).shortcutPushTo(&walk).shortcutDoublePushTo(&single_step);
-  stand_up.withRight(&sit_start).shortcutPushTo(&walk).shortcutDoublePushTo(&single_step);
-  sit_start.withRight(&obstacles).shortcutPushTo(&walk).shortcutDoublePushTo(&single_step);
-  obstacles.shortcutPushTo(&walk).shortcutDoublePushTo(&single_step);
-  walk.backTo(&obstacles).shortcutPushTo(&obstacles).shortcutDoublePushTo(&single_step);
-  single_step.backTo(&obstacles).shortcutDoublePushTo(&obstacles).shortcutPushTo(&walk);
+  home_stand_start.withRight(&home_sit_start).withLeft(&turn_off_start);
+  home_sit_start.withRight(&stand_up);
+  stand_up.withRight(&sit_start);
+  sit_start.withRight(&obstacles);
 
   // Set start state
   this->current_state_ = &home_sit_start;
 }
 
-void StateMachine::constructWalkMenu(State* from, State* prev_gait)
+void StateMachine::constructWalkMenu(std::list<State>::iterator last_state)
 {
   State& walk_small =
-      this->createGaitState(WALK_SMALL, WALK_SMALL_SELECTED, WALK_SMALL_ACTIVATED, "gait_walk_small", prev_gait);
+      this->createEscapeGaitState(WALK_SMALL, WALK_SMALL_SELECTED, WALK_SMALL_ACTIVATED, "gait_walk_small", nullptr);
   State& walk_normal =
-      this->createGaitState(WALK_NORMAL, WALK_NORMAL_SELECTED, WALK_NORMAL_ACTIVATED, "gait_walk", prev_gait);
-  State& walk_large = this->createState(WALK_LARGE);
+      this->createEscapeGaitState(WALK_NORMAL, WALK_NORMAL_SELECTED, WALK_NORMAL_ACTIVATED, "gait_walk", nullptr);
+  State& walk_large = this->createEscapeState(WALK_LARGE);
 
-  walk_small.backTo(from);
-  walk_normal.backTo(from).withLeft(&walk_small);
-  walk_large.backTo(from).withLeft(&walk_normal).withRight(&walk_small);
+  walk_small.withRight(&walk_normal);
+  walk_normal.withRight(&walk_large);
+  walk_large.withRight(&walk_small);
 
-  from->withSelect(&walk_normal);
+  std::list<State>::iterator state_iterator;
+  for (state_iterator = this->states_.begin(); state_iterator != last_state; state_iterator++)
+  {
+    if ((state_iterator->getGaitName()).empty())
```

Comment:
I think these brackets are redundant
```suggestion
    if (state_iterator->getGaitName().empty())
```

Reply:
Yes, they are redundant. Just made it easier to read for me, but I will remove them